rero_ils/modules/documents/views.py

```diff
@@ -637,3 +575,38 @@ def get_articles(record):
             'pid':hit.pid
         })
     return articles
+
+
+@blueprint.app_template_filter()
+def online_holdings(document_pid, viewcode='global'):
+    """Find holdings by document pid and viewcode.
+
+    : param document_pid: document pid
+    : param viewcode: symbol of organisation viewcode
+    : return: list of holdings
+    """
+    holdings = []
+    organisation = None
+    if viewcode != current_app.config.get('RERO_ILS_SEARCH_GLOBAL_VIEW_CODE'):
+        organisation = Organisation.get_record_by_viewcode(viewcode)
+    from ..holdings.api import HoldingsSearch
+    query = HoldingsSearch()\
+        .filter('term', document__pid=document_pid)
+    if organisation:
+        query = query.filter('term', organisation__pid=organisation.pid)
+    results = query.source(['library', 'electronic_location', 'notes']).scan()
+    holdings = []
+    for record in results:
+        library = Library.get_record_by_pid(record.library.pid)
+        record.library.name = library['name']
+        if 'notes' not in record:
+            record.notes = []
+        public_notes_content = [
+            n['content']
+            for n in record.notes
+            if n['type'] in [HoldingNoteTypes.PUBLIC]
+        ]
+        if public_notes_content:
+            record.notes = public_notes_content
```

Comment:
`HoldingNoteTypes.PUBLIC` doesn't exists !
L602 and L603 are not relevant because 'notes' are a property of holding record : remove it
If you want use the same structure as ItemNotes :
L610 you can't set value for a property (we don't define the setter method)
```python
                public_notes_content = [
                    n['content']
                    for n in record.to_dict().get('notes', [])
                    if n['type'] in HoldingNoteTypes.PUBLIC
                ]
                if public_notes_content:
                    record['notes'] = public_notes_content
```
Please check this code before merging